packages/flutter_tools/fuchsia_tools/README.md

```diff
@@ -0,0 +1,15 @@
+This directory serves as a placeholder directory for the package roots of
+various fuchsia tools which use the flutter_tools library.
+
+This is required to provide a workaround for the fuchsia build system.
+When this directory is not present the various tools specified in the
+`dart_tool` directives in the flutter_tools/BUILD.gn file will end up
+having the same package root entry written in the .packages file. This
```

Comment:
Why are they going into the same .packages file in the first place? Is the fix in the `dart_tool` template rather than here?

Reply:
It isn't so much that they are writing to the same .packages file but rather that we end up having two packages with the same packageUri. There was a recent change to the dart compiler which enforces the unique packageUri rule which is why the rolls into fuchsia have been failing. 

We need to update the dart build rules to use the new package_config.json format instead of the .packages format and I think we can figure out a more elegant solution here when that happens. We have had other places in the tree where this is a problem and we have had to restructure the code to fix the problem because we had something like a `flutter_app` and a `dart_library` in the same BUILD file.  

I am fine with not landing this change and just fixing the build rules to be more resilient but I'm not sure how long that will take and we have been without a roll on the order of ~2months now.